src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningDetailsExecutor.java

```diff
@@ -312,12 +318,12 @@ private String requestPropertyToken(TargetDeliveryRequest deliveryRequest) {
   }
 
   private boolean propertyTokenMismatch(List<String> rulePropertyTokens, String propertyToken) {
-    if (StringUtils.isEmpty(propertyToken)) {
-      return false;
-    }
     if (rulePropertyTokens == null || rulePropertyTokens.isEmpty()) {
       return false;
     }
```

Comment:
Why we have reordered the conditions?

Reply:
We need to check if a rule doesn't have any property tokens _before_ checking if there's no property supplied in the request, as if there's no property in the request we still need to get offers from rules that don't have a property token.